slick/src/main/scala/slick/jdbc/JdbcTypesComponent.scala

```diff
@@ -164,12 +173,185 @@ trait JdbcTypesComponent extends RelationalTypesComponent { self: JdbcProfile =>
+          */
+        java.sql.Types.VARCHAR
+      }
+      override def setValue(v: LocalTime, p: PreparedStatement, idx: Int) : Unit = {
+        p.setString(idx, v.toString)
+      }
+      override def getValue(r: ResultSet, idx: Int) : LocalTime = {
+        LocalTime.parse(r.getString(idx))
+      }
+      override def updateValue(v: LocalTime, r: ResultSet, idx: Int) = r.updateString(idx, v.toString)
+
+      override def valueToSQLLiteral(value: LocalTime) = {
+        value match {
+          case null => "NULL"
+          case _ => s"'${value.toString}'"
+        }
+      }
+    }
+
+    class LocalDateTimeJdbcType extends DriverJdbcType[LocalDateTime] {
+      // TODO: Use native Instant JDBC methods as soon they are implemented.
+      // https://java.net/jira/browse/JPA_SPEC-63
+      override def sqlType : Int = java.sql.Types.TIMESTAMP
+      override def setValue(v: LocalDateTime, p: PreparedStatement, idx: Int) : Unit = {
+        p.setTimestamp(idx, Timestamp.valueOf(v))
+      }
+      override def getValue(r: ResultSet, idx: Int) : LocalDateTime = {
+        r.getTimestamp(idx).toLocalDateTime
+      }
+      override def updateValue(v: LocalDateTime, r: ResultSet, idx: Int) : Unit = {
+        r.updateTimestamp(idx, Timestamp.valueOf(v))
+      }
+      override def hasLiteralForm = true
+      override def valueToSQLLiteral(value: LocalDateTime) = {
+        value match {
+          case null => "NULL"
+          case _ =>  s"'${Timestamp.valueOf(value)}'"
+        }
+      }
+    }
+
+    class InstantJdbcType extends DriverJdbcType[Instant] {
+      // TODO: Use native Instant JDBC methods as soon they are implemented.
+      // https://java.net/jira/browse/JPA_SPEC-63
+      override def sqlType : Int = {
+        java.sql.Types.TIMESTAMP
+      }
+      override def setValue(v: Instant, p: PreparedStatement, idx: Int) : Unit = {
+        p.setTimestamp(idx, Timestamp.from(v))
+      }
+      override def getValue(r: ResultSet, idx: Int) : Instant = {
+        r.getTimestamp(idx).toInstant
+      }
+      override def updateValue(v: Instant, r: ResultSet, idx: Int) : Unit = {
+        r.updateTimestamp(idx, Timestamp.from(v))
+      }
+      override def valueToSQLLiteral(value: Instant) = {
+        value match {
+          case null => "NULL"
+          case _ =>  s"'${Timestamp.from(value)}'"
+        }
+      }
+    }
+
+    class OffsetTimeJdbcType extends DriverJdbcType[OffsetTime] {
+      // TODO: Use native Instant JDBC methods as soon they are implemented.
+      // https://java.net/jira/browse/JPA_SPEC-63
+      override def sqlType : Int = {
+        /**
+          * Stores the [[OffsetTime]] as a 'VARCHAR' on databases, like SQLite, with no
+          * specific 'TIME' implementations. An example persisted value would be '18:22:43.417+01:00'.
+          * This is done because the data type 'TIME WITH TIMEZONE' is not available on all databases.
+          */
+        java.sql.Types.VARCHAR
+      }
```

Comment:
@cvogt 
I am trying to do this in this way, but every single database is failing, because they have implemented 'TIMESTAMP WITH TIMEZONE' in different ways, far away from the standard.

On the other side, most embedded databases do not support TIMESTAMPZ, other major databases, like MySQL, Microsoft SQL, DB2 do not support it, and there are some differences and limitations on the other databases, an example on how TIMESTAMP WITH TIMEZONE is implemented on some of the major relational databases can be found on the following link:

http://troels.arvin.dk/db/rdbms/

I would suggest implementing it generically as VARCHARS, and implement in the databases which support this (Oracle, PostgreSQL...) manually.